File: src/lib/objmdl/mdl/components/texture.ts
export interface Texture {
  id: number
  image: string;
  wrapWidth: boolean;
  wrapHeight: boolean;
}

export function texturesToString(textures: Texture[]) {
  if (textures.length === 0) return '';
  return `Textures ${textures.length} {
    ${textures.map((texture) => `
      Bitmap {
        Image "${texture.image}",
        ${texture.wrapWidth ? 'WrapWidth,' : ''}
        ${texture.wrapHeight ? 'WrapHeight,' : ''}
      }`).join('\n')}
  }`;
}

File: golang/internal/formats/mdl/components/texture.go
package components

import (
	"strings"
)

type TextureWowData struct {
	Type    int
	PngPath string
}

type Texture struct {
	ID            int
	ReplaceableID *int
	Image         string
	WrapWidth     bool
	WrapHeight    bool
	WowData       TextureWowData
}

func TexturesToString(textures []*Texture) string {
	if len(textures) == 0 {
		return ""
	}
	var b strings.Builder
	b.WriteString("Textures ")
	b.WriteString(FVal(float64(len(textures))))
	b.WriteString(" {\n")
	for _, texture := range textures {
		if texture == nil {
			continue
		}
		b.WriteString("Bitmap {\n")
		b.WriteString("Image \"")
		b.WriteString(strings.ReplaceAll(texture.Image, "/", "\\"))
		b.WriteString("\",\n")
		if texture.ReplaceableID != nil {
			b.WriteString("ReplaceableId ")
			b.WriteString(FVal(float64(*texture.ReplaceableID)))
			b.WriteString(",\n")
		}
		if texture.WrapWidth {
			b.WriteString("WrapWidth,\n")
		}
		if texture.WrapHeight {
			b.WriteString("WrapHeight,\n")
		}
		b.WriteString("}\n")
	}
	b.WriteString("}")
	return b.String()
}
